pkg/kubevirt/utils.go

```diff
@@ -156,17 +155,10 @@ func buildVirtualMachineInstanceTemplate(ctx *context.MachineContext) (*kubevirt
 	cloudInitType, index := detectCloudInitVolumeType(template)
 	cloudInitVolume := cloudinitVolume(template, ctx, cloudInitType)
 	cloudInitDisk := cloudinitDisk(template)
-	switch cloudInitType {
-	case cloudInitNoCloud:
-		err := mergo.Merge(&template.Spec.Volumes[index], cloudInitVolume)
-		if err != nil {
-			return nil, err
-		}
-		return template, nil
-	case cloudInitConfigDrive:
-		template.Spec.Volumes = append(template.Spec.Volumes, cloudInitVolume)
-	}
-	if !detectCloudInitDisk(template) {
```

Comment:
Just making sure it makes sense to remove this switch & merge logic with the simplified one below: